src/net/sourceforge/kolmafia/textui/javascript/ValueConverter.java

```diff
@@ -1,97 +1,38 @@
 package net.sourceforge.kolmafia.textui.javascript;
 
 import java.util.ArrayList;
-import java.util.Arrays;
 import java.util.List;
 import java.util.Map;
 import java.util.Map.Entry;
 import java.util.TreeMap;
 import net.sourceforge.kolmafia.MonsterData;
 import net.sourceforge.kolmafia.textui.DataTypes;
 import net.sourceforge.kolmafia.textui.DataTypes.TypeSpec;
-import net.sourceforge.kolmafia.textui.ScriptException;
 import net.sourceforge.kolmafia.textui.parsetree.AggregateType;
 import net.sourceforge.kolmafia.textui.parsetree.ArrayValue;
 import net.sourceforge.kolmafia.textui.parsetree.MapValue;
 import net.sourceforge.kolmafia.textui.parsetree.PluralValue;
 import net.sourceforge.kolmafia.textui.parsetree.RecordValue;
 import net.sourceforge.kolmafia.textui.parsetree.Type;
 import net.sourceforge.kolmafia.textui.parsetree.Value;
-import org.mozilla.javascript.ConsString;
-import org.mozilla.javascript.Context;
-import org.mozilla.javascript.EvaluatorException;
-import org.mozilla.javascript.NativeArray;
-import org.mozilla.javascript.NativeObject;
-import org.mozilla.javascript.Scriptable;
-import org.mozilla.javascript.ScriptableObject;
-import org.mozilla.javascript.Undefined;
 
-public class ValueConverter {
-  /* This is the closest thing that we have to undefined in our Value type system. Since javascript
-   * void functions return undefined, and we currently don't support undefined values as parameters,
-   * this is good enough for now.
-   */
-  private static final Type UNDEFINED_TYPE = new Type("undefined", TypeSpec.VOID);
-  private static final Value UNDEFINED =
-      new Value(UNDEFINED_TYPE) {
-        @Override
-        public String toString() {
-          return "undefined";
-        }
-      };
-
-  private final Context cx;
-  private final Scriptable scope;
-
-  public ValueConverter(Context cx, Scriptable scope) {
-    this.cx = cx;
-    this.scope = scope;
-  }
-
-  private Scriptable asObject(MapValue mapValue) {
-    Scriptable result = cx.newObject(scope);
-    for (Value key : mapValue.keys()) {
-      Value value = mapValue.aref(key);
-      if (key.getType().equals(DataTypes.STRING_TYPE)
-          || DataTypes.enumeratedTypes.contains(key.getType()) && key.contentString.length() > 0) {
-        ScriptableObject.putProperty(result, key.contentString, asJava(value));
-      } else if (key.getType().equals(DataTypes.INT_TYPE)
-          || DataTypes.enumeratedTypes.contains(key.getType()) && key.contentLong > 0) {
-        ScriptableObject.putProperty(result, (int) key.contentLong, asJava(value));
-      } else {
-        throw new ScriptException(
-            "Maps may only have keys of type string, int or an enumerated type.");
-      }
+public abstract class ValueConverter<ObjectType> {
+  public static class ValueConverterException extends RuntimeException {
```

Comment:
given how you've written the code (`throws`, translating this to `EvaluatorException`), why not make this a checked exception?